include/srf/pubsub/subscriber.hpp

```diff
@@ -0,0 +1,188 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "srf/codable/api.hpp"
+#include "srf/codable/decode.hpp"
+#include "srf/codable/encoded_object.hpp"
+#include "srf/control_plane/subscription_service_forwarder.hpp"
+#include "srf/node/edge_builder.hpp"
+#include "srf/node/operators/operator_component.hpp"
+#include "srf/node/queue.hpp"
+#include "srf/node/source_channel.hpp"
+#include "srf/pubsub/api.hpp"
+#include "srf/runtime/api.hpp"
+#include "srf/runtime/remote_descriptor.hpp"
+#include "srf/utils/macros.hpp"
+
+namespace srf::pubsub {
+
+/**
+ * @brief Receiving end of the Publisher->Subscriber chain
+ *
+ * Subscriber<T> is a source of T where T is received from the data plane as described in the data path below.
+ * Subscriber has a template specialization, i.e. Subscriber<RemoteDescriptor> which bypasses the decoding of the object
+ * and provides direct acess to the remote descriptor. This is useful for propagating the RemoteDescriptor without
+ * having to pull required bulk data held on the remote instance.
+ *
+ * Publisher<T> Data Path:
+ * [T] -> EncodedObject<T> -> EncodedStorage -> RemoteDescriptor -> Transient Buffer -> Data Plane Tagged Send
+ *
+ * Subscriber<T> Data Path:
+ * Data Plane Tagged Received -> Transient Buffer -> RemoteDescriptor -> Subscriber/Source<T> ->
+ *
+ * Subscriber<RemoteDescriptor> Data Path:
+ * Data Plane Tagged Received -> Transient Buffer -> RemoteDescriptor -> Subscriber/Source<RemoteDescriptor> ->
+ */
+template <typename T>
+class Subscriber final : public node::Queue<T>,
```

Comment:
If this inherits from `public node::Queue<T>` that means you can make connections that write to the subscriber. i.e. `node::make_edge(source, subscriber)`. This needs to inherit from something else I think.

Reply:
The new operator semantic should give us the option to make this a channel provider and not appear to have a dangling upstream connection. 

In fact, the upstrsm of the queue is connected to the internal pipeline that provides either a T or a RemoteDescriptkr.

Reply:
We can address after the non-linear pipeline stuff is merged.